Sources/TuistDependencies/Carthage/Utils/Carthage.swift

```diff
@@ -0,0 +1,94 @@
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+// MARK: - Carthage Command Generating
+
+/// Protocol that defines an interface to interact with the Carthage.
+public protocol Carthaging {
```

Comment:
Is there a reason why not to add this to `CarthageController`? IMHO `Carthaging` is a little bit awkward naming and not consistent with the rest of the codebase

Reply:
About naming: This is also similar to [GitHandler](https://github.com/tuist/tuist/blob/main/Sources/TuistSupport/Utils/GitHandler.swift) and `FileHandler` where we just call the system executables to perform tasks. We should align on a post-fix word for these, personally, I don't mind either but would be nice to add this to the `Contributing` section of the doc like a Tusit style guide 😎

Reply:
We also use `CarthageInteractor` and `SPMInteractor` which have the same concept of calling out to system executables.

Reply:
> Is there a reason why not to add this to CarthageController

OK, so today for `Carthage` we have `TuistSupport.CarthageController` [here](https://github.com/tuist/tuist/blob/main/Sources/TuistSupport/Carthage/CarthageController.swift)  and `TuistLoader.Carthage` [here](https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/Up/Carthage.swift). Both are used by `tuist up` feature. I believe `Carthage` support will be moved from `tuist up` to `tuist dependencies`. So to make it clean I wanted to have `TuistDependencies.*` owned objects to interact with dependencies managers because interaction with `Cocoapods`, `SPM` and `Carthage` will be possible only via `TuistDependencies.*` someday. (correct me if Im wrong). 

I can see that `TuistDependencies.CarthageInteractor` uses both `TuistDependencies.Carthage` and `TuistSupport.CarthageController`. It might be confusing. I'd remove `TuistSupport.CarthageController` from `TuistDependencies.CarthageInteractor`. What do you think? 

> IMHO Carthaging is a little bit awkward naming and not consistent with the rest of the codebase

Today we have `TuistLoader.Carthage` and `TuistLoader.Carthaging`. `FileHandling` and `GitHandling` are good examples too. So I can't agree it is not consistent with the codebase. An awkwardness is an other subject we can discuss here.

Reply:
The issue I have with `Carthage` is that it sounds like a model, rather than a service / dependency. I think we should aim for having single `CarthageController` and keep `CarthageInteractor` if necessary.